src/objects.rs

```diff
@@ -1286,6 +1286,11 @@ impl Calendar {
             dates: BTreeSet::new(),
         }
     }
+
+    /// Returns true if the calendars have at least one date in common
+    pub fn overlaps(&self, other: &Self) -> bool {
```

Comment:
I think it brings semantics to the operation and it will helps understand the code.
I'm not sure it really help to inline this (I don't really know the compiler policy on inlining), but if you think it's worth it, we can force the inlining here

Reply:
You're right for the 'semantic' part and I do agree with it. I'm resolving.